crates/miden-objects/src/errors.rs

```diff
@@ -403,6 +403,8 @@ pub enum NoteError {
     TooManyAssets(usize),
     #[error("note contains {0} inputs which exceeds the maximum of {max}", max = MAX_INPUTS_PER_NOTE)]
     TooManyInputs(usize),
+    #[error("note tag requires a public note but the note is of type {0:?}")]
+    PublicNoteRequired(NoteType),
```

Comment:
Let's remove `PublicUseCaseRequiresPublicNote` since this variant effectively replaces it. The usage of that variant in `impl TryFrom<MockChainNote> for InputNote` can be replaced with `PublicNoteRequired` as well.